src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -395,6 +395,12 @@ private BulkResponse execute() throws Exception {
                       + "will attempt retry after {} ms. Failure reason: {}",
                       batchId, batch.size(), attempts, maxAttempts, sleepTimeMs, e.getMessage());
             time.sleep(sleepTimeMs);
+            if (Thread.interrupted()) {
+              log.error(
+                      "Shutdown interrupted batch {} of {} records after attempt {}/{}",
```

Comment:
Also, adding the cause here seems too specific. Probably better to say something to signify that the thread that was waiting to retry a batch has been interrupted.

Reply:
I update the message to be a bit more generic and just talk about the retry being interrupted. PTAL?